File: src/System.Windows.Forms.Analyzers.VisualBasic/tests/UnitTests/System.Windows.Forms.Analyzers.VisualBasic.Tests/Analyzers/WFO1001/TestData/UnrelatedIDataObject.vb
Imports System
Imports System.Runtime.Versioning

Namespace System.Windows.Forms.Analyzers.VisualBasic.Tests.ImplementITypedDataObjectInAdditionToIDataObjectAnalyzer.Data

    Friend Interface IDataObject
        Sub DoStuff()
    End Interface

    Friend Class UnrelatedIDataObject
        Implements IDataObject

        Public Sub DoStuff() Implements IDataObject.DoStuff
            Throw New NotImplementedException()
        End Sub

    End Class

End Namespace

File: src/System.Windows.Forms.Analyzers.CSharp/tests/UnitTests/Analyzers/WFO1001/TestData/UnrelatedIDataObject.cs
using System;
using System.Runtime.Versioning;

namespace System.Windows.Forms.Analyzers.CSharp.Tests.ImplementITypedDataObjectInAdditionToIDataObjectAnalyzer.Data
{
    internal interface IDataObject
    {
        void DoStuff();
    }

    internal class UnrelatedIDataObject : IDataObject
    {
        public void DoStuff() => throw new NotImplementedException();
    }
}
